src/tests/test_utils/functional_test_utils/src/infer_model.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "functional_test_utils/infer_model.hpp"
+
+#include "common_test_utils/file_utils.hpp"
+#include "common_test_utils/test_constants.hpp"
+#include "functional_test_utils/ov_plugin_cache.hpp"
+#include "openvino/runtime/core.hpp"
+#include "openvino/util/file_util.hpp"
+#include "template/properties.hpp"
+
+namespace ov {
+namespace test {
+namespace utils {
+
+ov::TensorVector infer_on_template(const std::shared_ptr<ov::Model>& model, const ov::TensorVector& input_tensors) {
+    std::map<std::shared_ptr<ov::Node>, ov::Tensor> inputs;
+    auto params = model->inputs();
+    OPENVINO_ASSERT(params.size() == input_tensors.size());
+    for (int i = 0; i < params.size(); i++) {
+        inputs[params[i].get_node_shared_ptr()] = input_tensors[i];
+    }
+    return infer_on_template(model, inputs);
+}
+
+ov::TensorVector infer_on_template(const std::shared_ptr<ov::Model>& model,
+                                   const std::map<std::shared_ptr<ov::Node>, ov::Tensor>& inputs) {
+    ov::test::utils::PluginCache::get().reset();
```

Comment:
@olpipi 
Please use plugin cache, because `Template` plugin is registered here.
Also we will have a PR contains small refactoring for it. It will be only one place to create a core: **https://github.com/openvinotoolkit/openvino/pull/22595**

And please check the useful function in `calculate_refs`